driver/src/orderbook/streamed/balance.rs

```diff
@@ -2,17 +2,43 @@ use super::*;
 use anyhow::{anyhow, ensure, Result};
 use serde::{Deserialize, Serialize};
 
+/// Balance change from a potential solution that might still get replaced by a better solution in
+/// the same batch.
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+struct Proceeds {
+    batch_id: BatchId,
+    // Two fields instead of one I256 because although unlikely it could be possible to overflow a
+    // I256 with multiple trades in one batch.
+    increase: U256,
+    decrease: U256,
+}
+
+enum TradeType {
+    Sell,
+    Buy,
+}
+
+impl Proceeds {
+    fn get_field(&mut self, operation: TradeType) -> &mut U256 {
```

Comment:
nit: maybe an update proceeds function that takes a closure `Fn(batchId, amount) -> (new_batch_id, new_amount)` would allow a similar level of code-reuse without the need for TradeType and two fields. But this is just my personal taste, no good technical reason.

Reply:
I think the closure would fit nicely here.

Reply:
I don't understand this idea. We still need `increase` and `decrease` because of the overflow (unless we change to BigInt which I want to do later). If I get rid of `TradeType` but still have `increase` and `decrease` then how would the update function know which field to pass to the closure.

Reply:
I imagined it like this: `self.sell_buy_internal(amount, batch_id, |proceeds| &mut proceeds.increase)`, but I see that Felix has a alternate signature and maybe something slightly different in mind.